libtiny_client/src/state.rs

```diff
@@ -232,6 +271,31 @@ impl StateInner {
                 }
             }
 
+            // Reply 477 when user needs to be identified with NickServ to join a channel
+            Reply { num: 477, params } => {
+                // Only try to automatically rejoin if nickserv_ident is configured
+                if self.nickserv_ident.is_some() {
+                    // Get channel name
+                    let channel = params.get(1).unwrap().clone();
+                    // Find channel in self.chans
+                    let idx = utils::find_idx(&self.chans, |c| c.name == channel).unwrap();
+                    // Retry joining channel if retries are available
+                    if self.chans[idx].retry_join() {
+                        let mut snd_irc_msg = snd_irc_msg.clone();
+                        // Spawn task and delay rejoin to give NickServ time to identify nick
+                        tokio::task::spawn_local(async move {
+                            tokio::time::delay_for(tokio::time::Duration::from(
```

Comment:
This `Duration::from` is redundant as the argument is already a `Duration`.

Reply:
:woozy_face: Not sure how that happened...